lib/pages/chat/events/message.dart

```diff
@@ -106,6 +100,20 @@ class Message extends StatelessWidget {
           MessageTypes.Audio,
         }.contains(event.messageType);
 
+        final isPreviousEventMessage = {
+              EventTypes.Message,
+              EventTypes.Sticker,
+              EventTypes.Encrypted,
+              EventTypes.Redaction,
+            }.contains(previousEvent?.type);
+
+        final sameSender = previousEvent == null ||
```

Comment:
Can you break it down, this is hard to understand @@

Reply:
deserve a method

Reply:
@hoangdat @sherlockvn check [988cf68](https://github.com/linagora/twake-on-matrix/pull/302/commits/988cf68f9818cb0aa3a665d6031d0273e42749df) please